maybe_slugify() only does simple sanitation if config.slugify is false

southerntofu opened the pull request:
If this PR fits the problem, i will add some tests.

Discussion in #768

Commit:
maybe_slugify() only does simple sanitation if config.slugify is false

Commit:
slugify is disabled by default, turn on for backwards-compatibility

Commit:
First docs changes for optional slugification

Commit:
Remove # from slugs but not &

Commit:
Add/fix tests for utf8 slugs

Commit:
Fix test sites for i18n slugs

Commit:
fix templates tests for i18n slugs

Commit:
Rename slugify setting to slugify_paths

Commit:
Default slugify_paths

Commit:
Update documentation for slugify_paths

Commit:
Resolve conflict

Commit:
quasi_slugify removes ?, /, # and newlines

Keats commented on components/library/src/slugs.rs:
```diff
@@ -1,7 +1,7 @@
 
 pub fn quasi_slugify(s: &str) -> String {
-    s.replace("#", "").replace("/", "")
+    s.replace("#", "").replace("/", "").replace("?", "").replace("\n", "")
```
@southerntofu something I just remembered if that we should also removed invalid Windows characters in filenames like `:`.
It is already an issue in the page filename (https://www.getzola.org/documentation/content/page/): 

> The page name will be anything after _/- so a filename like 2018-10-10-hello-world.md will be available at [base_url]/hello-world. Note that the full RFC3339 datetime contains colons, which is not a valid character in a filename on Windows.

The full list: https://gist.github.com/doctaphred/d01d05291546186941e1b7ddc02034d3#file-ntfs-filenames-txt-L9-L17
Also needs to `trim_end` apparently: https://gist.github.com/doctaphred/d01d05291546186941e1b7ddc02034d3#file-ntfs-filenames-txt-L33

The full RFC3339 was already a mistake imo so let's not repeat it for actual pages/sections and be fully compatible with every OS.

southerntofu replied:
I think this is solved now. Also i updated the docs.

Commit:
Remove forbidden NTFS chars in quasi_slugify()

Commit:
Slugification forbidden chars can be configured

Commit:
Remove trailing dot/space in quasi_slugify

Commit:
Fix NTFS path sanitation

Commit:
Revert configurable slugification charset

Commit:
Remove \r for windows newlines and \t tabulations in quasi_slugify()

Commit:
Update docs for output paths

Keats commented on docs/content/documentation/content/page.md:
````diff
@@ -3,29 +3,48 @@ title = "Page"
+This is my article.
+```
+
+This frontmatter will output the article to `[base_url]/zines/femmes-libres-libération-kurde` with `slugify` disabled, and to `[base_url]/zines/femmes-libres-liberation-kurde` with `slugify` enabled.
````
slugify_path

Keats commented on docs/content/documentation/content/page.md:
```diff
@@ -3,29 +3,48 @@ title = "Page"
+
+If the path found starts with a datetime string (`YYYY-mm-dd` or [a RFC3339 datetime](https://www.ietf.org/rfc/rfc3339.txt)) followed by an underscore (`_`) or a dash (`-`), this date is removed from the output path and will be used as the page date (unless already set in the front-matter). Note that the full RFC3339 datetime contains colons, which is not a valid character in a filename on Window ...
+
+The output path extracted from the file path is then slugified or not depending on the `slugify` config, as explained previously.
```
slugify_path

Keats commented on docs/content/documentation/content/page.md:
```diff
@@ -3,29 +3,48 @@ title = "Page"
+- its `slug` frontmatter key
+- its filename
+
+Either way, these proposed path will be sanitized before being used. If `slugify_paths` is enabled in the site's config, paths are [slugified](https://docs.rs/slug/0.1.4/slug/fn.slugify.html). Otherwise, a simpler sanitation is performed, outputting only valid NTFS paths. The following characters are removed: `<`, `>`, `:`, `/`, `|`, `?` `*` `#` `\\`, as well as newlines and tabulations. Addition ...
```
I would link to wikipedia https://en.wikipedia.org/wiki/Clean_URL#Slug rather than the slug crate docs

Keats commented on test_site/config.toml:
```diff
@@ -4,6 +4,7 @@ highlight_code = true
 compile_sass = true
 generate_rss = true
 theme = "sample"
+slugify = true
```
slugify_path

Keats commented on CHANGELOG.md:
```diff
@@ -4,7 +4,8 @@
 
-- Pages with draft=true are now only loaded/rendered in `zola serve`
+- Add `--drafts` flag to `build`, `serve` and `check` to load drafts. Drafts are never loaded by default anymore
+- [Slugification](https://en.wikipedia.org/wiki/Slug_(web_publishing)#Slug) of page paths is now optional. By default, non-ASCII characters (international characters i.e. UTF8) will be retained in the generated pages. If you are currently using non-ASCII characters in your content file paths and want to keep the URLs Zola previously generated, set `slugify = true` in your config
```
slugify_path

Commit:
Replace slugify with slugify_paths

chrisvittal committed:
Reset color after writing message in colorize

Presently when you `^C` in `zola serve` it is painted with the same
color as the previous message. This PR always ensures to reset the color
in colorize, before writing the newline.

Keats committed:
Start 0.10

Keats committed:
Merge pull request #807 from chrisvittal/reset-color-properly

Reset color after writing message in colorize

Keats committed:
Looks like it now requires 1.36

Keats committed:
Merge branch 'next' into maybe_slugify
